cardano-cli/test/Test/Utilities.hs

```diff
@@ -1,27 +1,63 @@
-module Test.Utilities (diffVsGoldenFile) where
+module Test.Utilities
+  ( diffVsGoldenFile,
+    diffFileVsGoldenFile,
+  ) where
 
-import           Cardano.Prelude (ConvertText (..))
+import           Cardano.Prelude (ConvertText (..), HasCallStack)
 
 import           Control.Monad.IO.Class (MonadIO (liftIO))
 import           Data.Algorithm.Diff (PolyDiff (Both), getGroupedDiff)
 import           Data.Algorithm.DiffOutput (ppDiff)
 import           GHC.Stack (callStack)
+import qualified GHC.Stack as GHC
 import           Hedgehog (MonadTest)
+import qualified Hedgehog.Extras.Test as H
 import           Hedgehog.Extras.Test.Base (failMessage)
+import qualified Hedgehog.Internal.Property as H
+import qualified System.Directory as IO
+import qualified System.Environment as IO
+import qualified System.IO.Unsafe as IO
+
+-- | Whether the test should create the golden files if the file does ont exist.
+createFiles :: Bool
+createFiles = IO.unsafePerformIO $ do
+  value <- IO.lookupEnv "CREATE_GOLDEN_FILES"
+  return $ value == Just "1"
 
 -- TODO: Improve the help output by saying the difference of
 -- each input.
 diffVsGoldenFile
-  :: (MonadIO m, MonadTest m)
-  => String   -- ^ actual content
-  -> FilePath -- ^ reference file
+  :: HasCallStack
+  => (MonadIO m, MonadTest m)
+  => String   -- ^ Actual content
+  -> FilePath -- ^ Reference file
   -> m ()
-diffVsGoldenFile actualContent referenceFile =
-  do
-    referenceLines <- map toS . lines <$> liftIO (readFile referenceFile)
-    let difference = getGroupedDiff actualLines referenceLines
-    case difference of
-      [Both{}] -> pure ()
-      _        -> failMessage callStack $ ppDiff difference
+diffVsGoldenFile actualContent referenceFile = GHC.withFrozenCallStack $ do
+  fileExists <- liftIO $ IO.doesFileExist referenceFile
+
+  if fileExists
+    then do
+      referenceLines <- map toS . lines <$> H.readFile referenceFile
+      let difference = getGroupedDiff actualLines referenceLines
+      case difference of
+        [Both{}] -> pure ()
+        _        -> failMessage callStack $ ppDiff difference
+    else if createFiles
+      then do
+        H.note_ $ "Creating golden file " <> referenceFile
+        H.writeFile referenceFile actualContent
```

Comment:
So we will never fail in this case! We simply write the `actualContent` to the `referenceFile`.

Reply:
If the golden file already exists, it will not be overwritten.

Reply:
Ok, I understand what you've done. Can you add a comment that explains this case is simply to conveniently create golden files automatically.

Reply:
Comment added.